.github/workflows/build-publish.yml

```
@@ -0,0 +1,87 @@
+name: Build and publish Docker image with version
+on:
+  pull_request:
+    branches: [ master ]
+  push:
+    branches: [ master ]
+    tags:
+      - '*.*.*'
+  workflow_dispatch:
+
+jobs:
+  build_and_publish_docker:
+    name: Push Docker image to docker hub and github packages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.2.0
+        with:
+          fetch-depth: 0
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        id: buildx
+        with:
+          install: true
+      - name: 'Get Previous tag'
+        id: previoustag
+        uses: "WyriHaximus/github-action-get-previous-tag@v1"
+      - name: Prepare
+        id: prepare
+        run: |
+          function test_version() {
+            curl -s -S "https://registry.hub.docker.com/v2/repositories/openrouteservice/openrouteservice/tags/?page_size=1024" |
+              sed -e 's/,/,\n/g' -e 's/\[/\[\n/g' |
+              grep '"name"' |
+              awk -F\" '{print $4;}' |
+              sort -fu
+          }
+
+          DOCKER_IMAGE=openrouteservice/openrouteservice
+          CURRENT_VERSIONS=$(test_version)
+          DOCKER_VERSION=${{ steps.previoustag.outputs.tag }}
+          DOCKER_PLATFORMS=linux/amd64
+
+          TAGS_VERSION="--tag ${DOCKER_IMAGE}:${DOCKER_VERSION}"
+          TAGS_LATEST="--tag ${DOCKER_IMAGE}:latest"
+          TAGS_NIGHTLY="--tag ${DOCKER_IMAGE}:nightly"
+
+          if [[ $CURRENT_VERSIONS =~ $DOCKER_VERSION ]]; then
+            echo "Version: $DOCKER_VERSION present or latest. Skipping it!"
+            BUILD_VERSION=false
+          else
+            echo "Version $DOCKER_VERSION not present and not latest. Building it!"
```

I would rename this to IMAGE_VERSION or a more descriptive name, as this is not the docker version
```suggestion
          IMAGE_VERSION=${{ steps.previoustag.outputs.tag }}
          DOCKER_PLATFORMS=linux/amd64

          TAGS_VERSION="--tag ${DOCKER_IMAGE}:${IMAGE_VERSION}"
          TAGS_LATEST="--tag ${DOCKER_IMAGE}:latest"
          TAGS_NIGHTLY="--tag ${DOCKER_IMAGE}:nightly"

          if [[ $CURRENT_VERSIONS =~ $IMAGE_VERSION ]]; then
            echo "Image version: $IMAGE_VERSION present or latest. Skipping it!"
            BUILD_VERSION=false
          else
            echo "Image version $IMAGE_VERSION not present and not latest. Building it!"
```